[1671] Add course flow TDA changes

tomas-stefano opened the pull request:
### Context

We are now in a position to start building the TDA functionality for Publish. See the Trello card for what is in scope for this PR.

This PR deals with the add course logic changes to the add course multi step form.

<img width="908" alt="image" src="https://github.com/DFE-Digital/publish-teacher-training/assets/50492247/5b3b4b6c-4a01-4e34-99e6-62fdca76146e">


### Changes proposed in this pull request

- Add a new qualification for TDA
- Incorporate the new qualification in the Add course flow
- Skip questions and set defaults if the course is TDA (study mode, visa sponsorship and funding type)
- Remove change links on the Add course flow summary page for the above questions if the course is TDA

### Guidance to review

If you want to pull this down locally and review it on the UI, you have two options. 

1) You can rollover your local machine

2) you can turn the `teacher_degree_apprenticeship` feature flag on locally in `development.local.yml` and change your `current_recruitment_cycle_year` to `2025`. 

After doing one of the two options above, you should then be able to create a new course and see the `Teacher degree apprenticeship (TDA) with QTS` option under qualifications.

## Trello card

https://trello.com/c/fFK6beK0/1671-add-course-flow-tda-changes?filter=label:Squad%20B,label:Squad%20Unassigned

avinhurry commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add course flow TDA changes](https://trello.com/c/fFK6beK0/1671-add-course-flow-tda-changes)

elceebee commented on app/models/concerns/with_qualifications.rb:
```diff
@@ -75,7 +76,7 @@ def full_qualifications
       return '' unless qualifications
 
-      qualifications.map(&:upcase).sort.join(' with ')
+      I18n.t("qualifications.description.#{qualification}")
```
👍

elceebee commented on app/views/publish/courses/_basic_details_tab.html.erb:
```diff
@@ -75,7 +75,7 @@
        row.with_key { "Funding type" }
        row.with_value { course.funding }
-       if course.is_published? || course.is_withdrawn?
+       if course.is_published? || course.is_withdrawn? || course.teacher_degree_apprenticeship?
```
Will this need to change a year from now when TDA courses could be `draft_or_rolled_over? How will we remember to do that? I might be misunderstanding -- if so, it might be worth making a comment in the code as to why a tda course is being treated the same as a published or withdrawn course. The first (published and withdrawn) are states, where tda is type of course, so it's a bit confusing.

tomas-stefano replied:
Done

avinhurry replied:
Thanks @elceebee that's a great suggestion!

elceebee commented on app/views/publish/courses/confirmation.html.erb:
```diff
@@ -173,14 +177,16 @@
         <% end %>
 
-        <% if course.school_direct_salaried_training_programme? || course.pg_teaching_apprenticeship? %>
+        <% if course.school_direct_salaried_training_programme? || course.pg_teaching_apprenticeship? || course.teacher_degree_apprenticeship? %>
```
Is what makes these three things the same that they are salaried? or rather, do not have fees? A comment, or a descriptive method that returns the required courses might help future us know why we have this conditional

tomas-stefano replied:
Done

elceebee reviewed (commented):
I've tested it locally and I believe it works as expected. Just a few comments to help make the code a bit clearer for future us

tomas-stefano commented:
> I've tested it locally and I believe it works as expected. Just a few comments to help make the code a bit clearer for future us

Thank you for the review. I addressed your comments. :)

elceebee reviewed (commented):
Thanks for making the changes! It looks like maybe there have been some rebasing issues as there are changes in this that have already been merged.  Once that's fixed, I'll approve.
